src/call_state_machine.rs

```diff
@@ -0,0 +1,131 @@
+#[derive(Debug)]
+pub enum CallState {
+    Idle,
+    OutgoingCall,
+    IncomingCall,
+    InProgressCall,
+}
+
+#[derive(Debug)]
+pub enum CallSwitchEdge {
+    NoEdge,
+    Active,
+    Inactive,
+}
+
+pub enum CallSwitchState {
+    Active,
+    Inactive,
+}
+
+pub struct CallSwitch {
+    current_state: Option<CallSwitchState>,
+}
+
+impl CallSwitch {
+    fn new() -> Self {
+        CallSwitch {
+            //TODO read GPIO pin? Or set to None?
```

Comment:
I think starting with none and letting whoever calls dispatch read the gpio is better.

Reply:
yeah that's what I went with, can remove the comment